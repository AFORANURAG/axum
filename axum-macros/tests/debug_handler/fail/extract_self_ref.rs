use axum::{
    async_trait,
    extract::{FromRequest, RequestParts},
};
use axum_macros::debug_handler;

struct A;

#[async_trait]
impl<R, B> FromRequest<R, B> for A
where
    B: Send + 'static,
{
    type Rejection = ();

    async fn from_request(_req: &mut RequestParts<R, B>) -> Result<Self, Self::Rejection> {
        unimplemented!()
    }
}

impl A {
    #[debug_handler]
    async fn handler(&self) {}
}

fn main() {}
