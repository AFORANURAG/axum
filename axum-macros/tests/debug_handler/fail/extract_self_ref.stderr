error: Handlers must only take owned values
<<<<<<< HEAD
  --> tests/debug_handler/fail/extract_self_ref.rs:23:22
   |
23 |     async fn handler(&self) {}
=======
  --> tests/debug_handler/fail/extract_self_ref.rs:24:22
   |
24 |     async fn handler(&self) {}
>>>>>>> v0.7.0
   |                      ^^^^^
